src/sentry/incidents/models/incident.py

```diff
@@ -60,28 +60,39 @@ class Meta:
 
 class IncidentManager(BaseManager["Incident"]):
     CACHE_KEY = "incidents:active:%s:%s"
+    SUB_CACHE_KEY = "incidents:active:%s:%s:%s"
 
     def fetch_for_organization(self, organization, projects):
         return self.filter(organization=organization, projects__in=projects).distinct()
 
     @classmethod
-    def _build_active_incident_cache_key(cls, alert_rule_id, project_id):
+    def _build_active_incident_cache_key(cls, alert_rule_id, project_id, subscription_id):
+        if subscription_id:
+            return cls.SUB_CACHE_KEY % (alert_rule_id, project_id, subscription_id)
```

Comment:
When running a migration to add subscriptions to Incidents - how will this be affected?

we'll no longer find anything in the cache, and will build a new cache key with the subscription.
will there be any issue with the zombie cached items?
will there be any issue with querying for the incident once more?

Reply:
> will there be any issue with the zombie cached items?

There shouldn't be. You're not overridding the default TTL so keys will naturally expire in an hour. Memcache will reclaim the space used by old keys automatically.

Reply:
> will there be any issue with querying for the incident once more?
Querying for the incident will be fine too, it just means slightly higher db load until we populate the cache.